pulsar-broker/src/main/java/org/apache/pulsar/broker/service/EntryBatchIndexesAcks.java

```diff
@@ -50,12 +51,20 @@ public void recycle() {
         handle.recycle(this);
     }
 
+    private void ensureCapacityAndReset(int entriesListSize) {
+        size = entriesListSize;
+        if (indexesAcks.length < size) {
+            indexesAcks = new Pair[size];
+        } else {
+            for (int i = 0; i < size; i++) {
+                indexesAcks[i] = null;
```

Comment:
Instead of setting to null here, it would be better to do it in the recycle so the instances of `Pair` are staying valid for a smaller amount of time.

Reply:
Good point. I'll make this change.

Reply:
Done. PTAL @merlimat